src/s3ql/backends/gs.py

```diff
@@ -452,29 +487,14 @@ def write_fh(self, fh, key: str, md5: bytes,
         fh.seek(0)
 
         md5_run = hashlib.md5()
-        try:
-            self.conn.write(body_prefix)
-            while True:
-                buf = fh.read(BUFSIZE)
-                if not buf:
-                    break
-                self.conn.write(buf)
-                md5_run.update(buf)
-            self.conn.write(body_suffix)
-        except ConnectionClosed:
-            # Server closed connection while we were writing body data -
-            # but we may still be able to read an error response
-            try:
-                resp = self.conn.read_response()
-            except ConnectionClosed: # No server response available
-                pass
-            else:
-                if resp.status >= 400: # Got error response
-                    return resp
```

Comment:
@Nikratio (I think I accidentally deleted your comment and can't get it back.) The resp returned here is never used at the call site, that is why it didn't make sense to me.

Reply:
Good catch. The `return` here is completely bogus, presumably some leftover debugging code. Committed a fix.